librz/il/il_validate.c

```diff
@@ -529,6 +532,207 @@ VALIDATOR_PURE(loadw) {
 	return true;
 }
 
+VALIDATOR_PURE(float) {
+	RzILOpArgsFloat *args = &op->op.float_;
+	RzILSortPure sort;
+
+	if (!args->bv) {
+		rz_strbuf_appendf(report_builder, "Bitvector in constant float op is NULL.\n");
+		return false;
+	}
+
+	VALIDATOR_DESCEND(args->bv, &sort);
+	VALIDATOR_ASSERT(sort.type == RZ_IL_TYPE_PURE_BITVECTOR, "Float bv operand is not bitvector.\n");
+	*sort_out = rz_il_sort_pure_float(args->r);
+	return true;
+}
+
+VALIDATOR_PURE(fbits) {
+	RzILOpArgsFbits *args = &op->op.fbits;
+	RzILSortPure sort;
+
+	VALIDATOR_DESCEND(args->f, &sort);
+	VALIDATOR_ASSERT(sort.type == RZ_IL_TYPE_PURE_FLOAT, "operand of %s op is not a float.\n", rz_il_op_pure_code_stringify(op->code));
+	*sort_out = rz_il_sort_pure_bv(rz_float_get_format_info(sort.props.f.format, RZ_FLOAT_INFO_TOTAL_LEN));
+	return true;
+}
+
+// float -> bool
+VALIDATOR_PURE(float_bool_uop) {
+	RzILOpArgsIsFinite *args = &op->op.is_finite;
+	RzILSortPure sort;
+
+	VALIDATOR_DESCEND(args->f, &sort);
+	VALIDATOR_ASSERT(sort.type == RZ_IL_TYPE_PURE_FLOAT, "operand of %s op is not a float.\n", rz_il_op_pure_code_stringify(op->code));
+	*sort_out = rz_il_sort_pure_bool();
+	return true;
+}
+
+VALIDATOR_PURE(float_uop) {
+	RzILOpArgsFneg *args = &op->op.fneg;
+	RzILSortPure sort;
+
+	VALIDATOR_DESCEND(args->f, &sort);
+	VALIDATOR_ASSERT(sort.type == RZ_IL_TYPE_PURE_FLOAT, "operand of %s op is not a float.\n", rz_il_op_pure_code_stringify(op->code));
+	*sort_out = rz_il_sort_pure_float(sort.props.f.format);
+	return true;
+}
+
+VALIDATOR_PURE(fcast_to_int) {
+	RzILOpArgsFCastint *args = &op->op.fcast_int;
+	RzILSortPure sort;
+
+	VALIDATOR_DESCEND(args->f, &sort);
+	VALIDATOR_ASSERT(sort.type == RZ_IL_TYPE_PURE_FLOAT, "operand of %s op is not a float.\n", rz_il_op_pure_code_stringify(op->code));
+
+	*sort_out = rz_il_sort_pure_bv(args->length);
+	return true;
+}
+
+VALIDATOR_PURE(icast_to_float) {
+	RzILOpArgsFCastfloat *args = &op->op.fcast_float;
+	RzILSortPure sort;
+
+	VALIDATOR_DESCEND(args->bv, &sort);
+	VALIDATOR_ASSERT(sort.type == RZ_IL_TYPE_PURE_BITVECTOR, "operand of %s op is not a bitvector.\n", rz_il_op_pure_code_stringify(op->code));
+
+	*sort_out = rz_il_sort_pure_float(args->format);
```

Comment:
What about
```suggestion
	*sort_out = sort;
```
(also in similar places)

Reply:
Great, I will keep the same style